editor/src/components/canvas/grid-panels-state.tsx

```diff
@@ -0,0 +1,225 @@
+import findLastIndex from 'lodash.findlastindex'
+import { v4 as UUID } from 'uuid'
+import { accumulate, insert, removeAll, removeIndexFromArray } from '../../core/shared/array-utils'
+import { mod } from '../../core/shared/math-utils'
+import { assertNever } from '../../core/shared/utils'
+
+export const GridMenuWidth = 260
+export const GridPaneWidth = 500
+
+export const NumberOfColumns = 4
+export const IndexOfCanvas = 2
+
+export const GridPanelVerticalGapHalf = 6
+export const GridVerticalExtraPadding = 4
+export const GridPanelHorizontalGapHalf = 6
+export const GridHorizontalExtraPadding = 4
+
+export const ExtraHorizontalDropTargetPadding = 45
+
+export const GridPanelsNumberOfRows = 12
+
+export type Menu = 'inspector' | 'navigator'
+export type Pane = 'code-editor'
+
+export const allMenusAndPanels: Array<Menu | Pane> = [
+  'navigator',
+  'code-editor',
+  'inspector',
+  // 'preview', // Does this exist?
+]
+
+export interface GridPanelData {
+  panel: StoredPanel
+  span: number
+  index: number
+  order: number
+}
+
+export type PanelName = Menu | Pane
+
+export interface StoredPanel {
+  name: PanelName
+  type: 'menu' | 'pane'
```

Comment:
Since this only appears to be used for setting the width, should we just store the width here instead?

Reply:
nope! the width is a property of the column! this informs the default width (which is all we have since I haven't done the resizing yet) but it's not _the_ width which will be user changeable per column, not per panel item